batch/batch/server/server.py

```diff
@@ -452,8 +448,8 @@ def __init__(self, id, batch_id, attributes, callback, userdata, user, always_ru
         if self.is_complete():
             return
         if self._state == 'Created':
-            self._cancelled = True
-            await db.jobs.update_record(self.id, cancelled=True)
+            if not self.always_run:
+                await self.set_state('Cancelled')
         else:
             assert self._state == 'Ready', self._state
             await self.set_state('Cancelled')  # must call before deleting resources to prevent race conditions
```

Comment:
if the pod is already running, and we cancel it, even if it is always run, we still cancel the job. We should be consistent between the Ready and Created states.

Reply:
Actually, @cseed if we cancel a batch, that cancels all jobs, but should it delete the k8s resources for a running pod that is marked always_run?

Reply:
We should always run jobs that have been marked as always run even if the batch has been marked as cancelled.

Reply:
Then you agree this branch is wrong? Line 460/456 deletes all the k8s resources regardless of what `always_run` is set to.

Reply:
I agree it is wrong. Fixed.